### DISCUSSION TITLE
Allow to exclude hits on ToggleRefinement

### DISCUSSION THREAD

**Discussion by tristanbes** [algolia/instantsearch#4445] [General]
Hello,

I want to be able to exclude hits from the ToggleRefinement

```
As a user,
I want to mark items as "already viewed" from the search engine result;
I want to be able to remove those "already viewed hits" from the result by clicking on the ToggleRefinement.
```

Right now the ToggleRefinement only allows "include" and not "exclude".

Maybe a `mode` defaulting to "include", and allowing us to switch to "exclude" mode ?

PS: I tried playing with "on" and by prefixing hits by "-"

**Comment by Haroenv** [comment]
For me it works with a `-` in the value for "on": https://codesandbox.io/s/naughty-brahmagupta-p5bsi?file=/src/app.js, can you describe what you want exactly?

**Comment by Kocal** [comment]
Hey @Haroenv 👋 

I'm working with @tristanbes on his new feature, and we faced this issue where we can't exclude multiple values.

If I take your example, you are passing only one value `on: '-AT&T GoPhone'` and it works fine.
But if I use multiples values, eg: `on: ['-AT&T GoPhone', '-Boost Mobile']`, it does not work anymore. Please see https://codesandbox.io/s/fervent-dewdney-rv1cg?file=/src/app.js.

Do you have an idea of what can happens? Is it a limitation from Algolia itself or from instantsearch.js?

Thanks!

**Reply by Haroenv** [reply]
That's a limitation of the filter being used here (OR), if something is not GoPhone or not Boost mobile, it's actually everything again. ToggleRefinement always uses OR, but RefinementList has an option for AND

**Reply by Kocal** [reply]
Ah that's what I thought!

Would you accept a PR to add a property `operator` on `connectToggleRefinement` connector? Or it does not make sense?

**Reply by Haroenv** [reply]
It could make the connector quite complicated, I wonder if you're better off using the refine of a connectRefinementList, but if the impact is reasonable, that would definitely be interesting. The implementation would be swapping disjunctiveFacet with facet
